opentasks-provider/src/main/java/org/dmfs/provider/tasks/TaskDatabaseHelper.java

```diff
@@ -166,6 +168,40 @@
             + " JOIN " + Tables.LISTS + " ON (" + Tables.TASKS + "." + TaskContract.Tasks.LIST_ID + "=" + Tables.LISTS + "." + TaskContract.Tasks._ID + ")"
             + " JOIN " + Tables.INSTANCES + " ON (" + Tables.TASKS + "." + TaskContract.Tasks._ID + "=" + Tables.INSTANCES + "." + TaskContract.Instances.TASK_ID + ");";
 
+    /**
+     * SQL command to create a view that combines task instances with some data from the list they belong to. This replaces the task DTSTART, DUE and
+     * ORIGINAL_INSTANCE_TIME values with respective values of the instance.
+     * <p>
+     * This is the instances view as seen by the content provider clients.
+     */
+    private final static String SQL_CREATE_INSTANCE_CLIENT_VIEW = "CREATE VIEW " + Tables.INSTANCE_CLIENT_VIEW + " AS SELECT "
+            + Tables.INSTANCES + ".*, "
+            // override task due and start times with the instance values
+            + Tables.INSTANCES + "." + TaskContract.Instances.INSTANCE_START + " as " + Tasks.DTSTART + ", "
+            + Tables.INSTANCES + "." + TaskContract.Instances.INSTANCE_DUE + " as " + Tasks.DUE + ", "
+            // also replace ORIGINAL_INSTANCE_TIME.
```

Comment:
Why is it needed to replace ORIGINAL_INSTANCE_TIME? Doesn't it break the specification for that column on the contract? Its value will be different from what the docs says there, if I am not mistaken.

Reply:
No not really. `ORIGINAL_INSTANCE_TIME` is the time at which an instance occurs as per the specification in the master. The instances view represents a flattened view on the tasks, where each task instance is represented as a non-recurring override. Or in other words, it "emulates" an override for each instance of every (recurring or non-recurring) task.
So to me it makes sense to also "emulate" `ORIGINAL_INSTANCE_TIME` if present.

Reply:
This is still a bit confusing for me. Could you please remind me why the `Instances.INSTANCE_ORIGINAL_TIME` column is needed?

The docs for the two columns talk about almost the same things but with a bit different wording. Could it be maybe updated to be phrased the same? It could be easier to see the difference.

Reply:
I'm using `Instances.INSTANCE_ORIGINAL_TIME` primarily as a sorting key for all instanced of a recurring task. Without it I can't sort regular instances and overrides in the correct order. Correct sorting is very helpful when syncing the instances specified in the master with the instances in the database.
Also, with this column it is easier to handle recurring tasks without a start. In this case we assume `start == due` and store that as the `INSTANCE_ORIGINAL_TIME`. Without it, we would have to check at usage time which value we need to use, start or due (because we still store `null` in the start column).